source/FreeRTOS_TCP_Transmission_IPv4.c

```diff
@@ -234,19 +230,24 @@ void prvTCPReturnPacket_IPV4( FreeRTOS_Socket_t * pxSocket,
 
             pvCopySource = &pxEthernetHeader->xSourceAddress;
             ulDestinationIPAddress = pxIPHeader->ulDestinationIPAddress;
-            eARPLookupResult_t eResult;
 
-            eResult = eARPGetCacheEntry( &ulDestinationIPAddress, &xMACAddress, &( pxNetworkBuffer->pxEndPoint ) );
+            eResult = eARPGetCacheEntry( &ulDestinationIPAddress, &xMACAddress, &pxEndPoint );
 
             if( eResult == eARPCacheHit )
             {
                 pvCopySource = &xMACAddress;
+                pxNetworkBuffer->pxEndPoint = pxEndPoint;
             }
             else
             {
                 pvCopySource = &pxEthernetHeader->xSourceAddress;
             }
 
+            if( pxNetworkBuffer->pxEndPoint == NULL )
```

Comment:
Initially I had some confusion whether eARPGetCacheEntry will return a eARPCacheHit or eARPCacheMiss in this schenario and if the check (pxNetworkBuffer->pxEndPoint == NULL ) needs to be inside the return value check or outside as it is now. However, on some more exploration and internal discussions, it is clear that eARPGetCacheEntry will return eARPCacheMiss and the check can be outside and it will not have any other side-effect. With this I am good with the change.